vkcs/publicdns_record.go

```diff
@@ -0,0 +1,56 @@
+package vkcs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	recordTypeA     = "A"
+	recordTypeAAAA  = "AAAA"
+	recordTypeCNAME = "CNAME"
+	recordTypeMX    = "MX"
+	recordTypeNS    = "NS"
+	recordTypeSRV   = "SRV"
+	recordTypeTXT   = "TXT"
+)
+
+func publicDNSRecordStateRefreshFunc(client publicDNSClient, zoneID string, id string, recordType string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		record, err := recordGet(client, zoneID, id, recordType).Extract()
+
+		if err != nil {
+			if _, ok := err.(gophercloud.ErrDefault404); ok {
+				return record, recordStatusDeleted, nil
+			}
+			return nil, "", err
+		}
+
+		return record, recordStatusActive, nil
+	}
+}
+
+func publicDNSRecordParseZoneID(dns string) (string, error) {
+	dnsParts := strings.Split(dns, "/")
```

Comment:
a coment with dns string format whould be relevant

Reply:
Added: 
https://github.com/vk-cs/terraform-provider-vkcs/blob/0eb5e7947c41bb51fe453810073b0bb23655bf54/vkcs/publicdns_record.go#L37